src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
+import java.util.HashSet;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private HashSet<K> keys;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+        keys = new HashSet<K>();
+    }
+
+    public HashSet<K> keySet() {
+        return keys;
+    }
+
+    public boolean containsKey(K key) {
+        return keys.contains(key);
+    }
+
+    public V[] values() {
+        V[] values = (V[]) new Object[size];
+        return values;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (!containsKey(key)) {
+            add(key, value);
+        } else {
+            set(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (!containsKey(key)) {
+            return null;
+        } else {
+            return (V) getHashObject(key).value;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class HashObject<K, V> {
+        private K key;
+        private V value;
+        private HashObject<K, V> next;
+
+        public HashObject(K key, V value, HashObject<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private HashObject getHashObject(K key) {
+        HashObject currentNode = hashMap[keyIndex(key)];
```

Comment:
The `HashObject` should be parameterized with `<K, V>` when declaring `currentNode`.